### PR TITLE
split JSON API production setup doc to separate documents

### PR DESCRIPTION
**Pull request by S11001001**
Part of #15253, to make the production setup docs a little more manageable size-wise, especially as #15253 will grow them, and make things more linkable. Every top-level section in `json-api/production-setup` is now a separate page.

Running `git diff -C1 origin/main...origin/15253-split-production-setup-doc` locally will give you a much more sensible reviewable output than the GitHub tab here. The only actual changes that are not just moves are

1. add TOC to the root doc,
2. realign the headers to shift them one up, and
4. one typo fix: mimumum -> minimum

Please do not review any of the content that was moved; it will not be otherwise changed in this PR.

### CODE REVIEW

File: docs/source/json-api/production-setup/scaling-and-redundancy.rst [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+Therefore, if you do run with separate query stores, it may be useful to route queries
+(using a reverse proxy server) based on requesting party (and possibly queried template),
+which would minimize the amount of data in each query store as well as the overall
+redundancy of said data.
+
+Users may consider running PostgreSQL backend in a `high availability configuration <https://www.postgresql.org/docs/current/high-availability.html>`__.
+The benefits of this are use-case dependent as this may be more expensive for
+smaller active contract datasets, where re-initializing the cache is cheap and fast.
+
+Finally we recommend using orchestration systems or load balancers which monitor the health of
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/scaling-and-redundancy.rst:38]:
```suggestion
Finally, we recommend using orchestration systems or load balancers which monitor the health of
```

File: docs/source/json-api/production-setup/scaling-and-redundancy.rst [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+the service and perform subsequent operations to ensure availability. These systems can use the
+`healthcheck endpoints <https://docs.daml.com/json-api/index.html#healthcheck-endpoints>`__
+provided by the *HTTP JSON API* server. This can also be tied into supporting an arbitrary
+autoscaling implementation in order to ensure a minimum number of *HTTP JSON API* servers on
+failures.
+
+Set up the HTTP JSON API Service to work with Highly Available Participants
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/scaling-and-redundancy.rst:45]:
```suggestion
Set Up the HTTP JSON API Service To Work With Highly Available Participants
```

File: docs/source/json-api/production-setup/scaling-and-redundancy.rst [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+***************************************************************************
+
+In case the participant node itself is configured to be highly available, depending on the setup you may want
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/scaling-and-redundancy.rst:48]:
```suggestion
If the participant node itself is configured to be highly available, depending on the setup you may want
```

File: docs/source/json-api/production-setup/scaling-and-redundancy.rst [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+to choose different approaches to connect to the participant nodes. In most setups, including those based on Canton,
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/scaling-and-redundancy.rst:49]:
```suggestion
to choose different approaches to connect to the passive participant node(s). In most setups, including those based on Canton,
```
Is this what you meant?

File: docs/source/json-api/production-setup/security-and-privacy.rst [hunk 5]
```diff
@@ -0,0 +1,40 @@
+.. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Security and Privacy
+####################
+
+For an *HTTP JSON API* server, all data is maintained by the operator of the deployment.
+Thus, it is the operator's responsibility to ensure that the data contained abides by the necessary
+regulations and confidentiality expectations.
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/security-and-privacy.rst:9]:
```suggestion
It is the operator's responsibility to ensure that the data abides by the necessary
regulations and confidentiality expectations.
```

File: docs/source/json-api/production-setup/security-and-privacy.rst [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+
+We recommend using the tools documented by PostgreSQL to protect data at
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/security-and-privacy.rst:11]:
```suggestion
We recommend using the tools documented by PostgreSQL to protect data at
```
Ideally we would provide a link to that documentation here.

File: docs/source/json-api/production-setup/security-and-privacy.rst [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+rest, and using a secure communication channel between the *HTTP JSON API* server and the PostgreSQL server.
+
+To protect data in transit and over untrusted networks, the *HTTP JSON API* server provides
+TLS support. To enable TLS you need to specify both the private key for your server and the
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/security-and-privacy.rst:15]:
```suggestion
 The *HTTP JSON API* server provides
TLS support to protect data in transit and over untrusted networks. To enable TLS you must specify both the private key for your server and the
```

File: docs/source/json-api/production-setup/security-and-privacy.rst [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+        // trust collection, which means that all client certificates will be verified using the trusted
+        // certificates in this store. if omitted, the JVM default trust store is used.
+        trust-collection-file = "root-ca.crt"
+      }
+    }
+
+Using the cli options (deprecated), you can specify tls options using``daml json-api --pem server.pem --crt server.crt``.
+Custom root CA certificate can be set via ``--cacrt ca.crt``
+
+For more details on secure Daml infrastructure setup please refer to this `reference implementation <https://github.com/digital-asset/ex-secure-daml-infra>`__
```
Comment (carrielaben-da) [inline docs/source/json-api/production-setup/security-and-privacy.rst:39]:
```suggestion
For more details on secure Daml infrastructure setup please see this `reference implementation <https://github.com/digital-asset/ex-secure-daml-infra>`__
```

**Review by carrielaben-da — COMMENTED** [review]
Will these need a separate folder and index page in the TOC?

**Comment by S11001001** [comment]
> Will these need a separate folder and index page in the TOC?

@carrielaben-da I don't think so; `docs.daml.com` repo doesn't even refer to `production-setup`, and these new pages seem to be incorporated into the `docs/source/index.rst` TOC without any further changes. But, er, @garyverhaegen-da ?

File: docs/source/json-api/production-setup/query-store.rst [hunk 9]
```diff
@@ -0,0 +1,95 @@
+.. Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+.. SPDX-License-Identifier: Apache-2.0
+
+Query Store
+###########
+
+.. note:: Daml Open Source only supports PostgreSQL backends for the *HTTP JSON API* server, but Daml Enterprise also supports Oracle backends.
+
+The *HTTP JSON API* server is a JVM application that by default uses an in-memory backend.
```
[the remark this answers is not in the document]
Comment (S11001001) [reply docs/source/json-api/production-setup/query-store.rst:9]: 2f9ca495be3da3ef6ccfeeca8e4476e42946c35c

File: docs/source/json-api/production-setup/query-store.rst [hunk 10]
```diff
@@ -0,0 +1,95 @@
...
+This in-memory backend setup is inefficient for larger datasets as every query ends up fetching the entire active contract set for all the templates the query references. For this reason, for production setups, we recommend, at a minimum, that one use a database as a query store. This will allow for a more efficient caching of data and will improve query performance. Details for enabling a query store are highlighted below.
```
[the remark this answers is not in the document]
Comment (S11001001) [reply docs/source/json-api/production-setup/query-store.rst:10]: 2f9ca495be3da3ef6ccfeeca8e4476e42946c35c

File: docs/source/json-api/production-setup/query-store.rst [hunk 11]
```diff
@@ -0,0 +1,95 @@
...
+
+The ``start-mode`` is a custom parameter, defined by the query store configuration itself, which allows one to deal
+with the initialization and usage of the database backing the query store.
+
+Depending on how you prefer to operate it, you can either choose to:
+
+* run the *HTTP JSON API* server with ``start-mode=create-only`` with a user
+  that has exclusive table-creating rights that are required for the query store
+  to operate, and then start it once more with ``start-mode=start-only`` with a user
+  that can use the aforementioned tables, but that cannot apply schema changes
```
[the remark this answers is not in the document]
Comment (S11001001) [reply docs/source/json-api/production-setup/query-store.rst:65]: This has been reworked a bit more radically.

File: docs/source/json-api/production-setup/query-store.rst [hunk 12]
```diff
@@ -0,0 +1,95 @@
...
+
+Data Continuity
+***************
+
+The query store is a cache. This means that it is perfectly fine to drop it, as
+the data it contains is a subset of what can safely be recovered from the ledger.
+
+As such, the query store does not provide data continuity guarantees across versions
+and furthermore doesn't guarantee that a query store initialized with a previous
+version of the *HTTP JSON API* will be able to work with a newer version.
```
[the remark this answers is not in the document]
Comment (S11001001) [reply docs/source/json-api/production-setup/query-store.rst:82]: 2f9ca495be3da3ef6ccfeeca8e4476e42946c35c

File: docs/source/json-api/production-setup/query-store.rst [hunk 13]
```diff
@@ -0,0 +1,95 @@
...
+
+However, the *HTTP JSON API* is able to tolerate working with query stores initialized
+by a previous version of the software so long as the underlying schema did not change.
+
+The query store keeps track of the schema version under which it was initialized and
+refuses to start if a new schema is detected when it's run with a newer version.
+
+To evolve, the operator of the *HTTP JSON API* query store needs to drop the database
+used to hold the *HTTP JSON API* query store, needs to create a new one (consult your database
+vendor's documentation as to how this should be done), and then, depending on the operator's preferred production setup, should proceed to create and
```
[the remark this answers is not in the document]
Comment (S11001001) [reply docs/source/json-api/production-setup/query-store.rst:92]: 2f9ca495be3da3ef6ccfeeca8e4476e42946c35c

**Review by ray-roestenburg-da — APPROVED** [review]

File: docs/source/json-api/production-setup.rst [hunk 14]
```diff
@@ -6,397 +6,12 @@ Production Setup
...
-``daml.http_json_api.upload_packages_throughput``
--------------------------------------------------
-
-A meter. Number of package uploads
-
-``daml.http_json_api.allocation_party_throughput``
---------------------------------------------------
-
-A meter. Number of party allocations
+.. toctree::
```
Comment (garyverhaegen-da) [inline docs/source/json-api/production-setup.rst:9]: No tocs in this repo please, add them to [docs.daml.com](https://github.com/digital-asset/docs.daml.com).

Comment (S11001001) [reply docs/source/json-api/production-setup.rst:9]: What does this mean, when json-api/index has one?

Comment (S11001001) [reply docs/source/json-api/production-setup.rst]: eb464700025c2232c5316b845b4ab24476288964

Comment (S11001001) [reply docs/source/json-api/production-setup.rst]: digital-asset/docs.daml.com#55

File: docs/source/json-api/production-setup/metrics.rst [hunk 15]
```diff
@@ -0,0 +1,168 @@
...
+.. code-block:: none
+
+    metrics {
+      //Start a metrics reporter. Must be one of "console", "csv:///PATH", "graphite://HOST[:PORT][/METRIC_PREFIX]", or "prometheus://HOST[:PORT]".
+      reporter = "console"
+      //Set metric reporting interval , examples : 1s, 30s, 1m, 1h
+      reporting-interval = 30s
+    }
+
+or the two following CLI options (deprecated):
```
[the remark this answers is not in the document]
Comment (S11001001) [reply docs/source/json-api/production-setup/metrics.rst:22]: I expect them to be removed with Daml 3.

File: docs/source/json-api/production-setup/architecture.rst [hunk 16]
```diff
@@ -0,0 +1,26 @@
...
+
+- the *HTTP JSON API* server
+- the query store backend database server
+- the ledger
+
+The *HTTP JSON API* server exposes an API to interact with the Ledger and it uses JDBC to interact
+with its underlying query store in order to cache and serve data efficiently.
+
+The *HTTP JSON API* server releases are regularly tested with OpenJDK 11 on a x86_64 architecture,
+with Ubuntu 20.04, macOS 11.5.2, and Windows Server 2016.
```
[the remark this answers is not in the document]
Comment (S11001001) [reply docs/source/json-api/production-setup/architecture.rst:20]: 5b952894d3c

**Pull request merged**